src/instructlab/model/llmaaj.py

```diff
@@ -0,0 +1,388 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+import enum
+import logging
+import pathlib
+import os
+import yaml
+
+# Third Party
+import httpx
+import click
+from openai import OpenAI, OpenAIError
+import pandas as pd
+from ragas.callbacks import ChainRun
+from ragas.dataset_schema import EvaluationDataset, EvaluationResult
+from pydantic import (
+    BaseModel,
+    Field,
+    ValidationError,
+)
+
+# First Party
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator, RunConfig, Sample, _DEFAULT_SYSTEM_PROMPT
+from instructlab.eval.mt_bench_common import get_openai_client
+from instructlab import client_utils
+
+# Local
+from ..utils import is_model_gguf, is_model_safetensors, validate_taxonomy_file
+from ..client_utils import http_client, ClientException
+from .evaluate import launch_server, get_model_name
+from ..configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+class llmaaj_judge(BaseModel):
+    name: str = Field(
+        default='gpt-4o',
+        description="Judge model name for judge doing evaluation in the LLMaaJ evaluation. Model name must be an Open AI GPT model.",
+    )
+
+class llmaaj_model(BaseModel):
```

Comment:
so instead of having this as a separate config class in this pkg, I think this class should live in `configuration.py` and the flags in `evaluate.py` should inherit the defaults from the `config.yaml` as all other commands do. Introducing it this way would be a pretty big departure from what other commands are doing.

we just introduced `config_class=` which you can specify in your command line options like:

```
click.option(
"--llamaaj",
config_class="llamaaj",
...
```

meaning your flags can inherit defaults from other parts of the config besides `evaluate`